ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ConnectExecutor.java

```diff
@@ -59,7 +71,43 @@ private ConnectExecutor() { }
       );
     }
 
+    if (createConnector.ifNotExists() && response.httpCode() == HttpStatus.SC_CONFLICT) {
+      return Optional.of(new WarningEntity(statement.getStatementText(),
+          String.format("Connector %s already exists", createConnector.getName())));
+    }
```

Comment:
Should be sufficient to check if the connector now exists, e.g .

```suggestion
    if (createConnector.ifNotExists()) {
      // Check again to see if above call failed because someone else created the same connector:
      final Optional<KsqlEntity> connectorsResponse = handleIfNotExists(statement, createConnector, client);
      if (connectorsResponse.isPresent()) {
         return connectorsResponse.get();
      }
    }
```

Reply:
Yeah , have changed this to check if failure was because of connector being created again